Allow configuration of trailing commas in multi-line signatures (#12975)
sphinx-doc/sphinx: CHANGES.rst

Features added
--------------

* #12975: Add the possibility to avoid rendering a trailing comma in Python and
Javascript multi-line signatures, via
:confval:`python_trailing_comma_in_multi_line_signatures` and
:confval:`javascript_trailing_comma_in_multi_line_signatures`, respectively.

Bugs fixed
----------

> * #12975: Avoid rendering a trailing comma in C/C++ multi-line signatures.

Reviewer: ```suggestion
* #12975: Avoid rendering a trailing comma in C and C++ multi-line signatures.
```

Reviewer: The use of "C/C++" makes it sound like they are one thing, but they are two very different languages, and also domains in Sphinx.

Bugs fixed
----------

> * #12975: No trailing comma in C/C++ multi-line signatures

Reviewer 2: ```suggestion
* #12975: Avoid rendering a trailing comma in C/C++ multi-line signatures.
```

You can credit yourself if you want!

Features added
--------------

> * #12975: Add the possibility to avoid rendering a trailing comma in Python and
>   Javascript multi-line signatures, via
>   :confval:`python_trailing_comma_in_multi_line_signatures` and
>   :confval:`javascript_trailing_comma_in_multi_line_signatures`, respectively.

Reviewer 2: ```suggestion
* #12975: Add the possibility to avoid rendering a trailing comma in Python and
  Javascript multi-line signatures via
  :confval:`python_trailing_comma_in_multi_line_signatures` and
  :confval:`javascript_trailing_comma_in_multi_line_signatures` respectively.
```
